### DISCUSSION TITLE
General Question and Sugguestion

### DISCUSSION THREAD

**Discussion by seppy87** [nlohmann/json#2289] [Q&A] [answered]
Hallo,

I have a question:
I am developing an administration tool over network it passes Results and commands via JSON.
But I split the result (in case of big data in pieces)

json resp;
resp["action"] = [Index Number of Action]
resp["from"] = User Identifier
and there is an json object with result (e.g. listed all installed Devices with Informations about it)

for(auto& device : object){
resp[device["name"].get<std::string>()] = device;
FIFOBuffer.insert(resp);
}

now my question is: will device Object and resp object be copied as expected in C++ or will it contain the same value (like a reference to resp object?

And I realized if I put u16string and u32string as value it gets saved as Array of Numbers. How about including a function json["key"].toStdU16String() ?

**Comment by nlohmann** [comment, marked as answer]
Objects are copied into the JSON structure.

The library supports UTF-8. There is no function to convert between encodings.

**Reply by nlohmann** [reply]
Do you need further assistance?
